backend/src/s9pk/mod.rs

```diff
@@ -123,10 +123,18 @@ pub async fn pack(#[context] ctx: SdkContext, #[arg] path: Option<PathBuf>) -> R
         })
         .scripts({
             let script_path = path.join(manifest.assets.scripts_path()).join("embassy.js");
-            if script_path.exists() {
-                Some(File::open(script_path).await?)
-            } else {
-                None
+            let needs_script = manifest.package_procedures().any(|a| a.is_script());
+            let has_script = script_path.exists();
+            match (needs_script, has_script) {
```

Comment:
:thinking:  Neat. I kinda like this pattern, but if we got more than two, maybe having enums to try and prevent boolean blindness? 

```rs
match (needs_script, has_script) {
  (AreScripts, HasScripts) => ...,
  (AreScript, MissingScripts) => ...,
  (NoScripts, HasScripts) => ...,
  (NoScripts, MissingScripts) => None,
```